### PR TITLE
Chore/move to typescript

### PR DESCRIPTION
**Pull request by qballer**
I've migrated the repo to typescript, this is a big change I admit, and still needs more. This should be big chunck of the work. While tests/build pass there is still  a need to further test this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,11 +14,13 @@
...
+    "prepublish": "npm run build",
+    "build": "tsc -d",
     "start": "node --use_strict index.js",
     "test": "jest",
     "restart:dev": "node helpers/reset; npm start"
   },
-  "main": "index.js",
+  "main": "dist/index.js",
   "bin": {
     "greenpress-auth": "index.js"
```
Comment (davidmeirlevy) [inline package.json:23]: You should change this path too..

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                         /* Enable incremental compilation */
+    "target": "es5",                                /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
```
Comment (davidmeirlevy) [inline tsconfig.json:7]: Why es5?
We're using node 14..

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 2 above)
Comment (qballer) [reply tsconfig.json:7]: cool

File: helpers/init-util.js [hunk 3]
```diff
@@ -1,41 +0,0 @@
-/**
- * this file used to initiate basic data inside the authentication service
- */
-const User = require('mongoose').model('User')
-const config = require('../config')
-
-function init () {
-  const user = new User({
-    tenant: process.env.TENANT,
-    email: process.env.EMAIL || 'test@test.com',
```
[the remark this answers is not in the document]
Comment (qballer) [reply helpers/init-util.js:10]: are all the helpers like that? 
If they are shouldn't they be in bin or something ?

**Review by davidmeirlevy — APPROVED** [review]

**Pull request merged**